store/src/java/com/zimbra/cs/service/mail/SendMsg.java

```diff
@@ -282,6 +292,48 @@ public Element handle(Element request, Map<String, Object> context, QName respQn
            }
     }
 
+    /**
+     * ZBUG-2310
+     * When we send mail first time to a new email id, it new contact in 
+     * AUTO_CONTACTS folder, and than we modify contact display name, 
+     * then this method is updating the contact record, so that we can see 
+     * updated display name in emailed contacts after sending email.
+     * @param acct This is the Account object to initialise object
+     * @param zsc  This is the ZimbraSoapContext object to initialise object
+     * @param octxt This is the OperationContext object to initialise object
+     * @param mm This is the MimeMessage object to get Recipient list
+     * @param mbox This is the Mailbox object to call modify contact
+     * @return nothing
+     */
+    public void modifyContactName(Account acct, ZimbraSoapContext zsc, OperationContext octxt, MimeMessage mm, Mailbox mbox)
+        throws ServiceException, MessagingException {
+        Address[] sentAddresses = null;
+        int contactId = 0;
+        String displayName = null;
+        sentAddresses = mm.getAllRecipients();
+        for(int i=0; i<sentAddresses.length; i++) {
+            InternetAddress address = (InternetAddress)sentAddresses[i];
+            if(address != null) {
+                displayName = Provisioning.getInstance().getAccountByName(address.getAddress()).getDisplayName();
+                if(displayName != null) {
+                    ContactAutoComplete autoComplete = new ContactAutoComplete(acct, zsc, octxt);
+                    AutoCompleteResult result = autoComplete.query(address.getAddress());
+                    Collection<ContactEntry> entries = result.entries;
+                    for(ContactEntry con: entries) {
+                        contactId = con.getId().getId();
+                    }
+                ParsedContact pc = null;
+                HashMap<String, String> contactFields = new HashMap<>();
+                contactFields.put("firstName", displayName);
+                contactFields.put("fullName", displayName);
+                contactFields.put(MailConstants.A_EMAIL, address.getAddress());
+                pc = new ParsedContact(contactFields);
+                mbox.modifyContact(octxt, contactId, pc);
```

Comment:
should we add a check if the contactId is found only call modify else not?

Reply:
Done.